kv.go

```diff
@@ -344,8 +344,9 @@ const (
 
 // Regex for valid keys and buckets.
 var (
-	validBucketRe = regexp.MustCompile(`\A[a-zA-Z0-9_-]+\z`)
-	validKeyRe    = regexp.MustCompile(`\A[-/_=\.a-zA-Z0-9]+\z`)
+	validBucketRe    = regexp.MustCompile(`\A[a-zA-Z0-9_-]+\z`)
+	validKeyRe       = regexp.MustCompile(`\A[-/_=\.a-zA-Z0-9]+\z`)
+	validSearchKeyRe = regexp.MustCompile(`^[^ >]*[>]?$`)
```

Comment:
this would allow matching keys that arent valid keys like `x!y`

Reply:
You're right, I'll just base it on `validKeyRe` + allow wildcards